### DISCUSSION TITLE
Register stimulus controller during bundle install

### DISCUSSION THREAD

**Discussion by Sven-Ve** [symfony/symfony#42410] [Q&A]
I would like to register a Stimulus JavaScript controller during bundle installs (using composer).

Just like symfony/ux-chartjs does. So in _assets/controllers.json an_ entry like:
```json
{
    "controllers": {
        "@symfony/ux-chartjs": {
            "chart": {
                "enabled": true,
                "fetch": "eager"
            }
        }
    },
    "entrypoints": []
}
```

Unfortunately, no idea how to do that. There is no explanation in the recipes documentation.
For symfony/ux-chartjs I can't find a recipe either, so I could learn from it...

**Comment by javiereguiluz** [comment]
Maybe @tgalopin can help us here to confirm if this is possible or not to do it. Thanks!

**Comment by ker0x** [comment]
TBC, but I think you need do declare it in your `package.json`, based on https://github.com/symfony/ux/blob/main/src/Chartjs/Resources/assets/package.json#L6

**Reply by Sven-Ve** [reply]
I created a small test bundle and also created the package.json file there, see https://github.com/Sven-Ve/svc-test1-bundle.

But this does not help.  Neither `composer req svc/test1-bundle` nor `yarn install --force` makes the entry in controllers.json.

I don't know how symfony/ux-chartjs does it, since it doesn't even have a recipe....

**Comment by Sven-Ve** [comment]
ok, I have a version that works.
See https://github.com/Sven-Ve/svc-test2
The difference to test1 is that `Resources/assets` is directly in the root directory and not under `/src`

Also, the bundle is not called Bundle ;-)
But I don't know if it's because of that, I will create a 3rd version later on

**Comment by Sven-Ve** [comment]
ok, solution (??) found.

It needs two conditions:
- `Resources/assets` must be directly in the root directory.
- In `composer.json` it needs the following entry:

```json
  "keywords": [
    "symfony-ux"
  ],
```

But now I have no idea if I can trust this (undocumented?) symfony-ux entry for the future...???

see example: https://github.com/Sven-Ve/svc-test3-bundle

**Comment by Sven-Ve** [comment]
One last small correction:

As I found in the flex source code, `package.json` can be either in `/Resources/assets` (as described above) or directly in the `/asset`s directory

see https://github.com/symfony/flex/blob/main/src/PackageJsonSynchronizer.php#L210

But now I still do not know if I am allowed to use this
```json
  "keywords": [
    "symfony-ux"
  ],
```

**Reply by slaci** [reply]
According to the [flex source](https://github.com/symfony/flex/blob/v1.13.4/src/PackageJsonSynchronizer.php#L206) code you must use that `symfony-ux` keyword in your composer.json, otherwise flex won't even look for your package.json file (as of `v1.13.4`).

**Comment by lyrixx** [comment]
I has been implemented 🎉
